go/vt/orchestrator/collection/collection.go

```diff
@@ -283,7 +283,7 @@ func (c *Collection) Append(m Metric) error {
 	c.Lock()         //nolint SA5011: possible nil pointer dereference
 	defer c.Unlock() //nolint SA5011: possible nil pointer dereference
 	// we don't want to add nil metrics
-	if c == nil {
+	if c == nil { //nolint SA5011: redundant nil pointer check - maybe this was supposed to check if m == nil ?
```

Comment:
@shlomi-noach calling your particular attention to this line; I think this should be checking if `m == nil` (since we checked `c == nil` just before locking above) but wanted to confirm with you first.

Reply:
@ajm188 I believe you're right, this should be `m == nil`. Took me a while to understand why I have no recollection of this code, it was a contribution and I'm not too familiar with it.